### DISCUSSION TITLE
CSDN Massively Scrapes GitHub Projects to GitCode

### DISCUSSION THREAD

**Discussion by neo-turak** [community/community#130097] [Code Security]
### Select Topic Area

General

### Body

CSDN Massively Scrapes GitHub Projects to GitCode, Fakes Developer Profiles, Sparking Outrage

Incident Origin
CSDN's subsidiary, GitCode, has recently caused widespread anger and protests among developers due to an extremely unethical behavior. Without notifying or seeking the consent of developers, CSDN quietly transferred a large number of open-source projects from GitHub to its own GitCode platform and created developer profiles for these projects. This action not only infringes on the intellectual property and labor of developers but has also thoroughly enraged the open-source community.

Scraping Behavior
Initially, I thought there were many "scraper" users on the CSDN platform, but it turns out that CSDN itself is the true "king of scrapers"! They not only rudely transferred projects from GitHub but also replaced the GitHub links in the project README files with GitCode links.

What is even more shocking is that these scraped projects include proxy tools like v**ay and other sensitive projects. Although this inappropriate content has since been removed (resulting in 404 errors), it does not affect users from using gitclone to clone these projects, which is truly laughable.

Latest Developments

Currently, many people have noticed their projects being covertly transferred, and a large number of developers have flooded the GitCode official community demanding the complete removal of their projects from the platform.

However, because most developers have never registered with GitCode, they are completely unable to stop this scraping behavior. This means that even though the projects are transferred, the developers do not have control over them, despite their names still being associated with the projects.

Such behavior, which disregards developers' rights and maliciously transfers projects, is bound to receive strong condemnation from the developer community. Respecting developers' work and maintaining a positive atmosphere in the open-source community is crucial. Developers should unite to collectively resist this unethical behavior and protect their legitimate rights.

Github will do action for that or not?

At the end, please block the CSDN
(
*://*.csdn.net/*
*://*.gitcode.com/*
*://*.gitcode.net/*
*://*.gitcode.host/*
)

**Comment by StreamingMoon** [comment]
CSDN has heavily **violated the moral principles** of open-sourse spirit and **the Chinese law**. It is necessary to **take action**, or the heart of communication and sharing would be trampled. Most of us are waiting for Github's reply.

Here is some related law.
(Not translated, to avoid ambiguity)

中华人民共和国《计算机软件保护条例》
第十六条　软件的合法复制品所有人享有下列权利：
（一）根据使用的需要把该软件装入计算机等具有信息处理能力的装置内；
（二）为了防止复制品损坏而制作备份复制品。这些备份复制品不得通过任何方式提供给他人使用，并在所有人丧失该合法复制品的所有权时，负责将备份复制品销毁；
（三）为了把该软件用于实际的计算机应用环境或者改进其功能、性能而进行必要的修改；但是，除合同另有约定外，未经该软件著作权人许可，不得向任何第三方提供修改后的软件。

第二十三条　除《中华人民共和国著作权法》或者本条例另有规定外，有下列侵权行为的，应当根据情况，承担停止侵害、消除影响、赔礼道歉、赔偿损失等民事责任：
（一）未经软件著作权人许可，发表或者登记其软件的；
（二）将他人软件作为自己的软件发表或者登记的；
（三）未经合作者许可，将与他人合作开发的软件作为自己单独完成的软件发表或者登记的；
（四）在他人软件上署名或者更改他人软件上的署名的；
（五）未经软件著作权人许可，修改、翻译其软件的；
（六）其他侵犯软件著作权的行为。

第二十四条　除《中华人民共和国著作权法》、本条例或者其他法律、行政法规另有规定外，未经软件著作权人许可，有下列侵权行为的，应当根据情况，承担停止侵害、消除影响、赔礼道歉、赔偿损失等民事责任；同时损害社会公共利益的，由著作权行政管理部门责令停止侵权行为，没收违法所得，没收、销毁侵权复制品，可以并处罚款；情节严重的，著作权行政管理部门并可以没收主要用于制作侵权复制品的材料、工具、设备等；触犯刑律的，依照刑法关于侵犯著作权罪、销售侵权复制品罪的规定，依法追究刑事责任：
（一）复制或者部分复制著作权人的软件的；
（二）向公众发行、出租、通过信息网络传播著作权人的软件的；
（三）故意避开或者破坏著作权人为保护其软件著作权而采取的技术措施的；
（四）故意删除或者改变软件权利管理电子信息的；
（五）转让或者许可他人行使著作权人的软件著作权的。
有前款第一项或者第二项行为的，可以并处每件100元或者货值金额1倍以上5倍以下的罚款；有前款第三项、第四项或者第五项行为的，可以并处20万元以下的罚款。

**Comment by StreamingMoon** [comment]
BTW, some of the developers wrote "original updated at Github" in readme.But after gitcode cloned them, they'll turn that to "original updated at Gitcode"

**Comment by LiaoxyuCM** [comment]
GitCode is likely to move all repositories, but it must move projects that Stars is greater than or equal to 10
Please check and block it

**Reply by StreamingMoon** [reply]
I've made all my projects private in order to avoid that.(2 weeks ago,actually)

I'm sorry to open-source community, but it's necessary.

**Reply by LiaoxyuCM** [reply]
In gitcode.host searth your repos or issues see how many copied

This is one way

**Reply by LiaoxyuCM** [reply]
My repos has not been copied but i check it every two hours.

**Reply by StreamingMoon** [reply]
It seems they've hidden the search result.
